internal/common/conversion/path_converter.go

```diff
@@ -0,0 +1,150 @@
+package conversion
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+)
+
+type TPFSchema interface {
+	TypeAtTerraformPath(context.Context, *tftypes.AttributePath) (attr.Type, error)
+}
+
+type TPFSrc interface {
+	GetAttribute(context.Context, path.Path, any) diag.Diagnostics
+}
+
+// AttributePathValue retrieves the value for src (state/plan/config) @ attributePath with converted path.Path, schema is needed to get the correct types.XXX (String/Object/etc.)
+func AttributePathValue(ctx context.Context, diags *diag.Diagnostics, attributePath *tftypes.AttributePath, src TPFSrc, schema TPFSchema) (attr.Value, path.Path) {
+	convertedPath, localDiags := AttributePath(ctx, attributePath, schema)
+	if localDiags.HasError() {
+		diags.Append(localDiags...)
+		return nil, convertedPath
+	}
```

Comment:
```suggestion
	convertedPath, localDiags := AttributePath(ctx, attributePath, schema)
        diags.Append(localDiags...)
	if diags.HasError() {
		return nil, convertedPath
	}
```

I think this is better, except if you want to ignore warnings on purpose, similar for other cases

Reply:
Thanks. Changed in a2bac7ba2ea0c449d2169a233ddb77ea5aa66ca0